src/framework/theme/components/radio/radio-group.component.ts

```diff
@@ -171,4 +174,16 @@ export class NbRadioGroupComponent implements AfterContentInit, OnDestroy, Contr
   protected markRadiosForCheck() {
     this.radios.forEach((radio: NbRadioComponent) => radio.markForCheck());
   }
+
+  protected subscribeOnRadiosBlur() {
+    merge(...this.radios.map(radio => radio.blur))
+      .pipe(
+        takeWhile(() => this.alive),
+        // wait for focus to be applied to another radio in case it moved within radio group.
+        delay(100),
+        filter(() => !this.radios.some(radio => radio.hasFocus)),
```

Comment:
`delay(100)` in this case may lead to unexpected behavior. So, maybe, we may remove it?